File: python/day-4/palindrome.py

test_list = ["madam", "racecar", "hello", "noon"]

def get_palindrome(list):
    result = []
    for string in list:
        left = 0
        right = len(string) - 1
        palindrome = True
        
        while left < right:
            if string[left] == string[right]:
                palindrome = False
                break
            left += 1
            right -= 1
        
        result.append(palindrome)
    
    return result


print(get_palindrome(test_list))

File: java-script/day-4/Palindrome.java

public class PalindromeChecker {
    
    public static BooleanPalindrome(String) {
         result = new Array();
        
        for (String  : list) {
            int left = 0;
            int right = string.length() - 1;
            boolean  Palindrome = true;
            
            while (left < right) {
                if (string.charAt(left) == string.charAt(right)) {
                     Palindrome = false;
                    break;
                }
                left++;
                right--;
            }
            
            result.add(Palindrome);
        }
        
        return result;
    }


   